Add support for configuring the IAM role used for the application

ejholmes opened the pull request:
This should be considered experimental for the time being. It allows you to set the IAM role used for an application.

There's still some gotcha's with using this. The primary one being that, if you set the role, attached `emp run's` won't get the role applied, since it's handled by the ECS agent (might work if the ECS agent is running on the host? Need to test).

phobologic commented on scheduler/cloudformation/template.go:
```diff
@@ -235,6 +235,13 @@ func (t *EmpireTemplate) addTaskDefinition(tmpl *troposphere.Template, app *sche
+	// If provided in the app environment, this role will be used when
+	// running tasks.
+	var taskRole interface{}
+	if arn, ok := app.Env["TASK_ROLE_ARN"]; ok {
```
This seems like a 'meta' environment variable, should it maybe start with `EMPIRE_`?

ejholmes replied:
Yeah, there's a couple of other places where we decide how to build the template based on environment variables, and it would probably be better if we prefixed those with something (other ones are `ECS_TASK_DEFINITION` and `LOAD_BALANCER_TYPE`).

I'm game for deprecating those, and switching them to `EMPIRE_*` prefixed.

phobologic replied:
Yeah, maybe we should do that here, and then for those old ones honor the old way and the new prefixed way for a release before deprecating the old one or something.  To be fair, I think both of the 'old' cases are still pretty new, right?

phobologic commented:
👍 Pretty awesome, we should probably have some docs detailing this - AND the limitation around `emp run` for now.

phobologic commented:
Also, have you verified that this actually works? I hope it does, considering what we've heard :)

ejholmes commented:
Ok @phobologic, I think this is good to go. Did some testing locally, and updated the example cloudformation stack to run Empire using a role assigned to the task definition.

I also added a commit that adds a prefix to the env vars that the cloudformation backend looks for to enable experimental features, which are now:

* `EMPIRE_X_LOAD_BALANCER_TYPE`
* `EMPIRE_X_TASK_DEFINITION_TYPE`
* `EMPIRE_X_TASK_ROLE_ARN`

If they ever move  to a non-experimental state, they'll either be enabled by default, or we'll remove the `X` from it.

phobologic commented:
👍 lets get this out there, I'm excited to use it :)
